src/components/modals/Community/Community.tsx

```diff
@@ -145,6 +157,29 @@ const Community = ({ chainProfiles }: { chainProfiles: ChainProfile[] }) => {
                             {friendRequests.length === 0 && <FriendRequests />}
                         </>
                     )}
+                    <ConfirmationDialog
```

Comment:
Is there a reason for `ConfirmationDialog` to be placed all the way up in `Community`? Should be fine to just place it inside `CommunityUserCard` and avoid the extra state passing.

Reply:
It would also remove the need for extra handling to avoid flickering when `friendUpForRemoval` becomes `null` before the dialog is completely hidden.

Reply:
Hm yeah, that is an issue that I did not foresee. I updated the code now so that the code is inside the UserCard. In addition, there is now a confirm dialog for denying requests. I opted to kinda dupe the modals to avoid flickering and to only check the `UserRelationShipAction` once.